src/main/java/org/solid/testharness/http/SolidClient.java

```diff
@@ -111,20 +115,34 @@ public URI getAclUri(final HttpHeaders headers) {
         final List<Link> links = HttpUtils.parseLinkHeaders(headers);
         final Optional<Link> aclLink = links.stream()
                 .filter(link -> link.getRels().contains("acl") ||
-                        link.getRels().contains("http://www.w3.org/ns/solid/acp#accessControl")).findFirst();
+                        link.getRels().contains("http://www.w3.org/ns/solid/acp#accessControl"))
+                .findFirst();
         return aclLink.map(Link::getUri).orElse(null);
     }
 
-    public boolean createAcl(final URI url, final String acl) throws IOException, InterruptedException {
-        logger.debug("ACL: {} for {}", acl, url);
-        final HttpResponse<Void> response = client.put(url, acl, HttpConstants.MEDIA_TYPE_TEXT_TURTLE);
-        return HttpUtils.isSuccessful(response.statusCode());
+    public String getAclType(final URI aclUri) throws IOException, InterruptedException {
+        final HttpResponse<Void> response = client.head(aclUri);
+        final List<Link> links = HttpUtils.parseLinkHeaders(response.headers());
+        final Optional<Link> acpLink = links.stream()
+                .filter(link -> link.getRels().contains("type") &&
+                        "http://www.w3.org/ns/solid/acp#AccessControlResource".equals(link.getUri().toString()))
+                .findFirst();
+        return acpLink.isPresent() ? ACP_MODE : WAC_MODE;
```

Comment:
I wonder if we should also check for WAC, and then fail (for some value of fail) if neither is recognized?

Reply:
* Determining a server uses ACP is easy as the ACL resource has this 'type' link but WAC has no equivalent. The ACL links on a resource will end up the same soon so the only other way I can think of to explicitly determine WAC is being used is to open an ACL resource and parse it to check RDF type of an Authorization.
* The jar contains the generated code from the artefact generator. Now that it is open source I can look at building the code within this project but that is a task for later.
* Ref slashes - I noticed that I had some duplicated code ensuring slashes are handled properly in the config files and simply extracted it to a utility class

Reply:
* Ah, yes, that's true. It is being discussed in #191
* OK, we can live with a small jar for now